### PR TITLE
Summary observations are now matched against responses with nearest time (1 second tolerance)

### PR DESCRIPTION
**Pull request by eivindjahren**
Since summary files have precision loss of time, we need to include some
tolerance when matching responses to observations.

Also contains a workaround for storage not handling datetimes with
microseconds due to index overflow in netcdf3.
https://github.com/equinor/ert/issues/6952

The added test currently takes ~40s to run, but improvements to ensemble_evaluator startup overhead (which is being worked on) should fix this.

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [x] Make sure tests pass locally (after every commit!)

## When applicable
- [x] **When there are user facing changes**: Updated documentation
- [x] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: src/ert/config/summary_config.py [hunk 1]
```diff
@@ -30,19 +30,6 @@ def __post_init__(self) -> None:
     def read_from_file(self, run_path: str, iens: int) -> xr.Dataset:
         filename = self.input_file.replace("<IENS>", str(iens))
         _, keys, time_map, data = read_summary(f"{run_path}/{filename}", self.keys)
-
-        if self.refcase:
```
Comment (oyvindeide) [inline src/ert/config/summary_config.py:34]: Can remove this property?

File: tests/integration_tests/test_observation_times.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+def test_small_time_mismatches_are_ignored(
+    responses, observation, tmp_path_factory, std_cutoff, enkf_alpha
+):
+    tmp_path = tmp_path_factory.mktemp("summary")
+    (tmp_path / "config.ert").write_text(
+        dedent(
+            f"""
+            NUM_REALIZATIONS 2
+            ECLBASE CASE
+            SUMMARY FOPR
```
Comment (oyvindeide) [inline tests/integration_tests/test_observation_times.py:59]: Not needed because of the SUMMARY_OBSERVATION (but can be here)

File: tests/integration_tests/test_observation_times.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+            MAX_SUBMIT 1
```
Comment (oyvindeide) [inline tests/integration_tests/test_observation_times.py:60]: Why 1?

File: tests/integration_tests/test_observation_times.py [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+                tmp_path / f"simulations/realization-{i}/iter-{j}/CASE.SMSPEC"
+            )
+            unsmry.to_file(
+                tmp_path / f"simulations/realization-{i}/iter-{j}/CASE.UNSMRY"
+            )
+    (tmp_path / "observations.txt").write_text(str(observation))
+
+    stderr = StringIO()
+    with redirect_stderr(stderr):
+        run_cli(ES_MDA_MODE, str(tmp_path / "config.ert"), "--weights=0,1")
```
Comment (oyvindeide) [inline tests/integration_tests/test_observation_times.py:91]: If you want to do this faster, it is possible to mock this by doing `load_from_forward_model` combined with a manual update.

File: tests/unit_tests/test_load_forward_model.py [hunk 5]
```diff
@@ -69,48 +69,6 @@ def run_simulator(time_step_count, start_date) -> Summary:
     return summary
 
 
-@pytest.mark.usefixtures("copy_snake_oil_case_storage")
-def test_load_inconsistent_time_map_summary(caplog):
```
Comment (oyvindeide) [inline tests/unit_tests/test_load_forward_model.py:73]: Is this covered by the new tests? Seems a bit unrelated?

File: tests/unit_tests/test_load_forward_model.py [hunk 6]
```diff
@@ -69,48 +69,6 @@ def run_simulator(time_step_count, start_date) -> Summary:
...
-    run_path = Path("storage") / "snake_oil" / "runpath" / "realization-0" / "iter-0"
-    os.chdir(run_path)
-    ecl_sum = run_simulator(1, datetime(2000, 1, 1))
-    ecl_sum.fwrite()
-    os.chdir(cwd)
-
-    realizations = [False] * facade.get_ensemble_size()
-    realizations[realisation_number] = True
-    with caplog.at_level(logging.WARNING):
-        loaded = facade.load_from_forward_model(ensemble, realizations, 0)
```
Comment (oyvindeide) [inline tests/unit_tests/test_load_forward_model.py:104]: This approach plus a manual update is possible, and would be a lot faster than run_cli.

File: src/ert/config/summary_config.py (hunk 1 above)
Comment (eivindjahren) [reply src/ert/config/summary_config.py:34]: Then we have to change storage?

File: tests/integration_tests/test_observation_times.py (hunk 3 above)
Comment (eivindjahren) [reply tests/integration_tests/test_observation_times.py:60]: Just used so that we don't retry (saves time).

File: tests/unit_tests/test_load_forward_model.py (hunk 5 above)
Comment (eivindjahren) [reply tests/unit_tests/test_load_forward_model.py:73]: It only checked that you got a warning that we don't expect to get any more.

File: tests/unit_tests/test_load_forward_model.py (hunk 6 above)
Comment (eivindjahren) [reply tests/unit_tests/test_load_forward_model.py:104]: Kind of hesitate to do add too touch too much detail in the test.

File: tests/integration_tests/test_observation_times.py (hunk 4 above)
Comment (oyvindeide) [reply tests/integration_tests/test_observation_times.py:91]: Not too many calls no, but adds complexity yes, not sure it is worth the trade-off, but because it is a property test it might be.

File: tests/unit_tests/test_load_forward_model.py (hunk 6 above)
Comment (eivindjahren) [reply tests/unit_tests/test_load_forward_model.py:104]: Tried doing this: 
```
def run_update(ert: LibresFacade) -> None:
    n = ert.get_ensemble_size()
    storage = open_storage(ert.enspath, mode="w")
    experiment = storage.create_experiment(
        parameters=list(ert.config.ensemble_config.parameter_configs.values())
    ).id
    prior = storage.create_ensemble(experiment=experiment, ensemble_size=n)
    posterior = storage.create_ensemble(experiment=experiment, ensemble_size=n)

    loaded = ert.load_from_forward_model(prior, np.array([True] * n), 0)
    assert loaded == n
    ert.smoother_update(prior, posterior, "")
```
but there were prior.has_data() fails.

Comment (eivindjahren) [reply tests/unit_tests/test_load_forward_model.py:104]: I think you need to do create_run_path and also have an ensemble_context. I suggest we just drop it and call run_cli for now.

File: src/ert/config/summary_config.py (hunk 1 above)
Comment (eivindjahren) [reply src/ert/config/summary_config.py:34]: I will create an issue for it

File: src/ert/config/_read_summary.py [hunk 7]
```diff
@@ -316,7 +316,8 @@ def _read_spec(
                         hour=hour,
                         minute=minute,
                         second=microsecond // 10**6,
-                        microsecond=microsecond % 10**6,
+                        # Due to https://github.com/equinor/ert/issues/6952
```
Comment (oyvindeide) [inline src/ert/config/_read_summary.py:319]: Could you make this comment stand-alone? Otherwise I foresee a future where the below line will simply be deleted, as I assume that it is related to this?

File: src/ert/config/_read_summary.py [hunk 8]
```diff
@@ -427,7 +441,14 @@ def read_params() -> None:
...
             vals = _check_vals("PARAMS", summary, last_params.read_array())
             values.append(vals[indices])
-            dates.append(start_date + unit.make_delta(float(vals[date_index])))
+
+            dates.append(
+                _round_to_seconds(
+                    start_date + unit.make_delta(float(vals[date_index]))
+                ),
+            )
+            # Due to https://github.com/equinor/ert/issues/6952
```
Comment (oyvindeide) [inline src/ert/config/_read_summary.py:450]: Could you make this comment stand-alone? Otherwise I foresee a future where the below line will simply be deleted, as I assume that it is related to this?

**Review by oyvindeide — APPROVED** [review]

File: src/ert/config/_read_summary.py (hunk 7 above)
Comment (eivindjahren) [reply src/ert/config/_read_summary.py:319]: Fixed

File: src/ert/config/_read_summary.py (hunk 8 above)
Comment (eivindjahren) [reply src/ert/config/_read_summary.py:452]: Fixed

File: src/ert/config/summary_config.py (hunk 1 above)
Comment (eivindjahren) [reply src/ert/config/summary_config.py:34]: Issue here: https://github.com/equinor/ert/issues/7015
